File: vkatz-lib/src/main/java/by/vkatz/utils/FontsManager.java
package by.vkatz.utils;

import android.content.Context;
import android.graphics.Typeface;

import java.util.HashMap;

/**
 * User: Katz
 * Date: 06.02.14
 * Time: 19:52
 */
@SuppressWarnings("WeakerAccess")
public class FontsManager {
    private static FontsManager _this;
    private HashMap<String, Typeface> fonts = new HashMap<>();

    private FontsManager() {
    }

    public static FontsManager instance() {
        if (_this == null) _this = new FontsManager();
        return _this;
    }

    public Typeface getFont(Context context, String fontName) {
        if (fontName == null || fontName.equals("")) return null;
        try {
            if (fonts.containsKey(fontName)) return fonts.get(fontName);
            else {
                Typeface font = Typeface.createFromAsset(context.getAssets(), fontName);
                fonts.put(fontName, font);
                return font;
            }
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: vkatz-lib/src/main/kotlin/by/vkatz/utils/FontsManager.kt
package by.vkatz.utils

import android.content.Context
import android.graphics.Typeface
import java.util.*

class FontsManager private constructor() {
    companion object {
        val instance: FontsManager by lazy { FontsManager() }
    }

    private val fonts = HashMap<String, Typeface>()

    fun getFont(context: Context, fontName: String?): Typeface? {
        if (fontName == null || fontName == "") return null
        return try {
            if (fonts.containsKey(fontName)) fonts[fontName]
            else {
                val font = Typeface.createFromAsset(context.assets, fontName)
                fonts.put(fontName, font)
                font
            }
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }
}
